wallet/wallet.go

```diff
@@ -5520,6 +5520,23 @@ func (w *Wallet) SetPublished(ctx context.Context, hash *chainhash.Hash, publish
 	return nil
 }
 
+// VSPTicketInfo returns the various information for a given vsp ticket
+func (w *Wallet) VSPTicketInfo(ctx context.Context, ticketHash *chainhash.Hash) (*udb.VSPTicket, error) {
+	data := &udb.VSPTicket{}
```

Comment:
This should be declared as `var data *udb.VSPTicket` to avoid allocating a value that is ultimately discarded (due to the next `data, err = udb.GetVSPTicket(..` always replacing this value.

Also, I notice this is returning a struct from the `udb` package, which I know @jrick wants to ultimately make an internal package, so maybe this is going to need to export an adapted struct instead (he can probably chime in on whether this should be done in this PR or left for a future work).

Reply:
Yah good point about the struct.  Added to the wallet package, maybe it makes more sense somewhere if you guys would like.